cmd/terramate/cli/run.go

```diff
@@ -143,6 +145,11 @@ func (c *cli) runOnStacks() {
 		c.detectCloudMetadata()
 	}
 
+	if c.parsedArgs.Run.CloudSyncPreview && os.Getenv("GITHUB_ACTIONS") == "" {
+		printer.Stderr.Warn(cloudSyncPreviewGHAWarning)
+		c.disableCloudFeatures(errors.E(cloudSyncPreviewGHAWarning))
+	}
```

Comment:
the `createCloudPreview()` is missing a guard for the case cloud is disabled.

Reply:
I am not sure if I understand, could you please link to the `createCloudPreview` where the guard is missing? 🤔

Reply:
just saw that it's handled in a `if` in another place:
```
if c.parsedArgs.Run.CloudSyncPreview && c.cloudEnabled() {
 ...
```

this is annoying because we are handling those cases in different ways..

for deployments, its handled inside the `createDeployment()`..
```
func (c *cli) createCloudDeployment(deployRuns []stackCloudRun) {
	// Assume each task in deployRuns has the CloudSyncDeployment set.
	logger := log.With().
		Logger()

	if !c.cloudEnabled() {
		return
	}
```

and for `drift` it's handled in the `cloudSyncAfter()`..

but okay, all good!